lib/internal/http2/core.js

```diff
@@ -717,6 +719,17 @@ function setupHandle(socket, type, options) {
 
   this[kHandle] = handle;
 
+  if (socket.encrypted) {
+    this[kAlpnProtocol] = socket.alpnProtocol;
+    this[kEncrypted] = true;
```

Comment:
Should we define these on the initial `class` (as `undefined`) to avoid thrashing of the hidden class? (Maybe `originSet` too, on `kState`.)